### PR TITLE
Fix signup continuation bug

### LINKED ISSUE #1777: AttributeError: 'SignupFlow' object has no attribute 'token'
Sentry Issue: [BACKEND-2B](https://sentry.io/organizations/couchers/issues/2507366629/?referrer=github_integration)

```
AttributeError: 'SignupFlow' object has no attribute 'token'
  File "couchers/interceptors.py", line 226, in tracing_function
    res = prev_func(request, context)
  File "couchers/interceptors.py", line 150, in user_unaware_function
    return user_aware_function(req, context)
  File "couchers/servicers/auth.py", line 163, in SignupFlow
    send_signup_email(existing_flow)
  File "couchers/tasks.py", line 25, in send_signup_email
    signup_link = urls.signup_link(token=flow.token)
```

### PR DESCRIPTION
**Pull request by aapeliv**
Closes #1777 (when done).

**Backend checklist**
- [ ] Formatted my code by running `autoflake -r -i --remove-all-unused-imports src && isort . && black .` in `app/backend`
- [x] Added tests for any new code or added a regression test if fixing a bug
- [x] All tests pass
- [ ] Run the backend locally and it works
- [x] Added migrations if there are any database changes, rebased onto `develop` if necessary for linear migration history

### CODE REVIEW

File: app/backend/src/tests/test_auth.py [hunk 1]
```diff
@@ -779,4 +779,44 @@ def test_complete_signup(db):
         assert e.value.details() == errors.INVALID_COORDINATE
 
 
+def test_signup_token_regression(db):
+    # Repro steps:
+    # 1. Start a signup
+    # 2. Confirm the email
+    # 3. Start a new signup with the same email
+    # Expected: send a link to the email to continue signing up.
+    # Actual: aforementioned error message.
```
Comment (kalvdans) [inline app/backend/src/tests/test_auth.py]: No error mentioned before. Say AttributeError.

File: app/backend/src/couchers/servicers/api.py [hunk 2]
```diff
@@ -33,7 +33,7 @@
 )
 from couchers.resources import language_is_allowed, region_is_allowed
 from couchers.sql import couchers_select as select
-from couchers.tasks import send_friend_request_email, send_report_email, send_friend_request_accepted_email
+from couchers.tasks import send_friend_request_accepted_email, send_friend_request_email, send_report_email
```
Comment (kalvdans) [inline app/backend/src/couchers/servicers/api.py]: Revert unrelated change

Comment (aapeliv) [reply app/backend/src/couchers/servicers/api.py]: This is just from isort and black, develop hadn't been formatted.

I can move it into another PR if you want though?

Comment (kalvdans) [reply app/backend/src/couchers/servicers/api.py]: Yes with a link to the offending PR that was merged despite not being formatted

Comment (aapeliv) [reply app/backend/src/couchers/servicers/api.py]: I'll make a separate PR

**Review by kalvdans — APPROVED** [review]
great with test!
